From 346fca6b74c4c2a627a05f182a5d49746067cc53 Mon Sep 17 00:00:00 2001
From: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Date: Tue, 2 Feb 2016 19:53:50 +0530
Subject: [PATCH 10/24] usb_pd: provide interface to enable/disable vconn thru
 gpio.

Currently vconn supply to fusb is by default always enabled and
fusb will enable/disable vconn to port partner based connection
state and pd negotiation. But  fusb chip expects the vconn supply to
the chip also to be enable/disable before enabling/disabling the vconn
to its port partner. This patch provides interface to enable/disable
vconn gpio which provides vconn supply fusb chip.

Change-Id: Id2ac387723f843f6bad0fa47fb38ed682b4a864d
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-18527
Signed-off-by: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Reviewed-on: https://android.intel.com:443/465550
---
 drivers/misc/intel_wcove_gpio.c | 17 ++++++++++++-----
 1 file changed, 12 insertions(+), 5 deletions(-)

diff --git a/drivers/misc/intel_wcove_gpio.c b/drivers/misc/intel_wcove_gpio.c
index 0d7e6bd..9bee55e 100644
--- a/drivers/misc/intel_wcove_gpio.c
+++ b/drivers/misc/intel_wcove_gpio.c
@@ -120,6 +120,17 @@ static int wcgpio_update_vbus_state(struct wcove_gpio_info *info, bool state)
 	return ret;
 }
 
+int wcgpio_set_vconn_state(bool state)
+{
+	if (!wc_info)
+		return -EINVAL;
+	gpiod_set_value_cansleep(wc_info->gpio_vconn, state);
+	dev_info(&wc_info->pdev->dev, "%s: vconn=%d\n",
+						__func__, state);
+	return 0;
+}
+EXPORT_SYMBOL(wcgpio_set_vconn_state);
+
 int wcgpio_set_vbus_state(bool state)
 {
 	int ret;
@@ -129,9 +140,9 @@ int wcgpio_set_vbus_state(bool state)
 
 	/* enable/disable vbus based on the provider(source) event */
 	if (!ret) {
+		gpiod_set_value_cansleep(wc_info->gpio_otg, state);
 		dev_info(&wc_info->pdev->dev, "%s: VBUS=%d\n",
 						__func__, state);
-		gpiod_set_value_cansleep(wc_info->gpio_otg, state);
 	}
 	return ret;
 }
@@ -317,11 +328,7 @@ static int wcove_gpio_probe(struct platform_device *pdev)
 		goto error_gpio;
 	}
 	dev_dbg(&pdev->dev, "wcove gpio probed\n");
-
 	check_initial_events(info);
-
-	/* Enable vconn always to typec chip */
-	gpiod_set_value_cansleep(info->gpio_vconn, 1);
 	wc_info = info;
 
 	return 0;
-- 
1.9.1

